tools/ipptransform.c

```diff
@@ -50,6 +50,26 @@ static inline fz_matrix make_matrix(float a, float b, float c, float d, float e,
  * Local types...
  */
 
+typedef void * (*renderer_make_renderer_cb_t)();
+typedef void (*renderer_deallocate_cb_t)(void *);
+typedef bool (*renderer_open_document_cb_t)(CFURLRef, void *);
+typedef int  (*renderer_get_page_count_cb_t)(void *);
+typedef bool (*renderer_load_page_cb_t)(const int, void *);
+typedef CGRect (*renderer_get_page_rect_cb_t)(void *);
+typedef CGAffineTransform  (*renderer_get_page_transform_cb_t)(void *);
+typedef bool (*renderer_render_cb_t)(CGContextRef, void *);
```

Comment:
What is the last argument of type `void *` in all of these?

Reply:
`void *` are pointers of unknown type.
It is the equivalent of an `UnsafeMutableRawPointer` in swift.

Int this case it ist the Swift `PdfKitRenderer` instance.
https://github.com/pickware/ios-cups-client/pull/26/files#diff-b19e0be84a257f6bab681b09bacf48fdR8

C has no concept of classes and instances, this is how we bridge c-functions to Swift classes, by passing the instance into each function.